### PR TITLE
Add support for RCODE and other header flag decoding.

### PR DESCRIPTION
**Pull request by pusateri**

### CODE REVIEW

File: rcodes.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    case 'NOTAUTH': return 9
+    case 'NOTZONE': return 10
+    case 'RCODE_11': return 11
+    case 'RCODE_12': return 12
+    case 'RCODE_13': return 13
+    case 'RCODE_14': return 14
+    case 'RCODE_15': return 15
+  }
+  return 0
+}
```
Comment (silverwind) [inline rcodes.js]: Need a trailing newline here.

**Comment by silverwind** [comment]
Is this done yet? It seems to be missing tests and the linter is failing.

**Comment by pusateri** [comment]
Sorry, still learning. I will fix the lint issue and add a test case.

**Comment by silverwind** [comment]
I'd suggest you use [editorconfig](http://editorconfig.org/). With it, things like trailing newlines will be handled automatically in by your editor.

**Comment by pusateri** [comment]
Ok, I think I have handled all of your concerns. Travis-ci seems to be having trouble right now with the node 0.12 version because of a key error unrelated to my changes though.

**Comment by silverwind** [comment]
Hmm, this just tests the conversion. Do you think it would be possible to test the properties of a decoded header? (Sorry, it's been a while since I've worked on this module)

**Comment by silverwind** [comment]
Thanks for the test. I've compared the output to do and would have to suggestions:

#### dig output
````
$ dig soa google.com

; <<>> DiG 9.9.7-P3 <<>> soa google.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 62316
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;google.com.      IN  SOA

;; ANSWER SECTION:
google.com.   53  IN  SOA ns1.google.com. dns-admin.google.com. 181011351 900 900 1800 60
````

#### dns-socket with dns-packet from your branch
````
> require('.')().query({
...   questions: [{
.....     type: 'SOA',
.....     name: 'google.com'
.....   }]
... }, 53, '8.8.8.8', console.log)
33463
> null { id: 33463,
  type: 'response',
  flags: 384,
  opcode: 0,
  flag_auth: 0,
  flag_trunc: 0,
  flag_rd: 1,
  flag_ra: 1,
  flag_z: 0,
  flag_ad: 0,
  flag_cd: 0,
  rcode: 'NOERROR',
  questions: [ { name: 'google.com', type: 'SOA', class: 1 } ],
  answers:
   [ { name: 'google.com',
       type: 'SOA',
       class: 1,
       ttl: 59,
       flush: false,
       data: [Object] } ],
  authorities: [],
  additionals: [] } { questions: [ { type: 'SOA', name: 'google.com' } ],
  type: 'query',
  flags: 256,
  id: 33463,
  answers: [],
  authorities: [],
  additionals: [] } 53 '8.8.8.8'
````

1. The `qr` flag that dig displays is missing, should it be added?
2. I'd change 0/1 in the flags to true/false which seems more appropriate in a language like JavaScript to me.

**Comment by silverwind** [comment]
Oh, and

3. I'd suggest resolving `opcode` to a string e.g. `0` for `'QUERY'`, like dig does it.

**Comment by pusateri** [comment]
1. The QR flag is currently showed as the type field (was there before I made any changes). But I can also add a boolean in addition.
2. I will change the integers to booleans.
3. Do I continue creating separate files for translating opcode values to strings like types.js and rcodes.js?

To be clear, your requests are inconsistent in format (even if they are consistent with dig). You are asking me to add a boolean for QR flag from a string that already exists and to change the integer value of opcode to a string.

**Comment by pusateri** [comment]
By the way, why do commits for pull request #14 show up here in pull request #15 ?

**Comment by silverwind** [comment]
> The QR flag is currently showed as the type field (was there before I made any changes). But I can also add a boolean in addition.

Looking at the RFC, I think it's actually fine the way you have it with the `type` string property. It seems more natural then having a boolean indicate query vs. response, so lets keep it.

> Do I continue creating separate files for translating opcode values to strings like types.js and rcodes.js?

Yeah, I think we should keep that tradition.

> By the way, why do commits for pull request #14 show up here in pull request #15 ?

Because you're making the common novice mistake of issueing pull requests from your `master` branch. I strongly advice to create separate feature branches for every pull request you make, so you can manage them independently. Also see [here](http://codeinthehole.com/tips/pull-requests-and-other-good-practices-for-teams-using-github/).

**Comment by pusateri** [comment]
Ok. Do you want me to now remove the Q/R boolean from my commit above or leave them both in?

**Comment by silverwind** [comment]
Hmm, I guess it won't hurt to keep it because it's technically a flag like the others. Let's keep it in.

**Comment by pusateri** [comment]
Closing this pull request. Will reopen a new one on a separate branch.

**Pull request closed without merging**